File: MafiaGame/Role.swift
import SwiftUI

enum Role: String, CaseIterable, Codable, Identifiable {
    case mafia = "Mafia"
    case crewmate = "Crewmate"
    case doctor = "Doctor"
    case investigator = "Investigator"
    case jester = "Jester"
    
    var id: String { rawValue }
    
    var icon: String {
        switch self {
        case .mafia: return "person.fill.viewfinder"
        case .crewmate: return "person.fill"
        case .doctor: return "cross.case.fill"
        case .investigator: return "magnifyingglass"
        case .jester: return "theatermasks.fill"
        }
    }
    
    var color: Color {
        switch self {
        case .mafia: return .red
        case .crewmate: return .blue
        case .doctor: return .green
        case .investigator: return .yellow
        case .jester: return .purple
        }
    }
    
    var team: Team {
        switch self {
        case .mafia: return .mafia
        case .crewmate, .doctor, .investigator: return .crew
        case .jester: return .neutral
        }
    }
    
    var description: String {
        switch self {
        case .mafia: return "Kill crewmates at night"
        case .crewmate: return "Find and eliminate the mafia"
        case .doctor: return "Save one player each night"
        case .investigator: return "Investigate one player each night"
        case .jester: return "Get voted out to win"
        }
    }
    
    enum Team: String {
        case mafia = "Mafia"
        case crew = "Crew"
        case neutral = "Neutral"
    }
}

File: web-app/src/models/Role.js
// Role model — mirrors iOS Role.swift exactly
export const RoleTeam = {
  MAFIA: 'mafia',
  CREW: 'crew',
  NEUTRAL: 'neutral',
};

export const Roles = {
  mafia: {
    id: 'mafia',
    name: 'Mafia',
    icon: '🔫',
    emoji: '🔫',
    color: '#ff3b30',
    team: RoleTeam.MAFIA,
    shortDescription: 'Eliminate crew members',
    description: 'Eliminate crew members each night. Win when Mafia equals or outnumbers the crew.',
    minRequired: 1,
    order: 0,
  },
  detective: {
    id: 'detective',
    name: 'Detective',
    icon: '🔍',
    emoji: '🔍',
    color: '#007aff',
    team: RoleTeam.CREW,
    shortDescription: 'Investigate suspicious players',
    description: 'Each night, investigate one player to learn if they are Mafia or not.',
    minRequired: 0,
    order: 1,
  },
  doctor: {
    id: 'doctor',
    name: 'Doctor',
    icon: '💉',
    emoji: '💉',
    color: '#34c759',
    team: RoleTeam.CREW,
    shortDescription: 'Save one player each night',
    description: 'Each night, choose one player to protect. If the Mafia targets them, they survive.',
    minRequired: 0,
    order: 2,
  },
  jester: {
    id: 'jester',
    name: 'Jester',
    icon: '🃏',
    emoji: '🃏',
    color: '#af52de',
    team: RoleTeam.NEUTRAL,
    shortDescription: 'Get yourself voted out to win',
    description: 'Win the game by getting yourself voted out during the day. You have no team.',
    minRequired: 0,
    order: 3,
  },
  civilian: {
    id: 'civilian',
    name: 'Civilian',
    icon: '👤',
    emoji: '👤',
    color: '#8e8e93',
    team: RoleTeam.CREW,
    shortDescription: 'Find and eliminate the Mafia',
    description: 'Participate in discussions and vote to eliminate suspected Mafia members.',
    minRequired: 0,
    order: 4,
  },
};

export function getRoleById(id) { return Roles[id] || Roles.civilian; }
export function getAllRoles() { return Object.values(Roles).sort((a, b) => a.order - b.order); }
export function getConfigurableRoles() { return getAllRoles(); }
